app/js/views/dev/camera-latest-image.jsx

```diff
@@ -10,36 +10,34 @@ export default class CameraLatestImage extends BaseView {
       service: null,
       hasPreview: false
     };
+
     this.foxbox = props.foxbox;
   }
 
   componentDidMount() {
-    this.foxbox.getService(this.props.id).then((response) => {
-      const service = response.data;
-
-      this.setState({ service });
+    this.foxbox.getService(this.props.id)
+      .then((service) => {
+        this.setState({ service });
 
-      return this.foxbox.performGetOperation(
-        this.getGetOperation(service, 'latest image')
-      );
-    })
-    .then((image) => {
-      this.refs.snapshotPreview.src = URL.createObjectURL(image);
-      this.setState({ hasPreview: true });
-    })
-    .catch((e) => {
-      console.error(
-        'Error occurred while retrieving latest image for camera (id=%s): ',
-        this.props.id,
-        e
-      );
-    });
+        return this.state.service.takeSnapshot();
```

Comment:
__potential issue:__ iirc I read the react doesn't refresh state immediately, I'd say it would be safe to use ```service``` here directly